src/norm_oracles.jl

```diff
@@ -360,3 +360,61 @@ function compute_extreme_point(lmo::EllipsoidLMO, direction; v=nothing, kwargs..
     mul!(v, I, lmo.buffer, -scaling, true)
     return v
 end
+
+
+"""
+    OrderWeightNormLMO(weights,radius)
+    
+LMO with feasible set being the atomic ordered weighted l1 norm: https://arxiv.org/pdf/1409.4271
+
+```
+C = {x ∈ R^n, Ω_w(x) ≤ R} 
+```
+
+"""
+struct OrderWeightNormLMO{W,R,P,B,D} <: LinearMinimizationOracle
+    weights::W
+    radius::R
+    perm_indices::P
+    mat_B::B
+    direction_abs::D
+end
+
+function OrderWeightNormLMO(weights, radius)
+    N = length(weights)
+    mat = zeros(N,N)
+    s = zero(eltype(weights))
+    for i in 1:N
+        s += weights[i]
+        for j in 1:i
+            mat[i,j] = 1 / s
+        end
+    end
+    perm_indices = sortperm(weights,rev=true)
+    direction_abs = similar(weights)
+    return OrderWeightNormLMO(weights, radius, perm_indices, mat, direction_abs)
+end
+
+function compute_extreme_point(
+    lmo::OrderWeightNormLMO{W,R,P,B,D},
+    direction::M;
+    v=similar(direction),
+    kwargs...,
+) where {W,R,P,B,D,M}
```

Comment:
```suggestion
    lmo::OrderWeightNormLMO,
    direction::M;
    v=similar(direction),
    kwargs...,
)
```

types not needed if you don't use them